src/integration-test/java/com/mongodb/jdbc/integration/testharness/DataLoader.java

```diff
@@ -42,13 +42,15 @@
 
     private List<TestDataEntry> datasets;
     private Set<Pair<String, String>> collections;
+    private Set<Pair<String, String>> views;
```

Comment:
What is the purpose of this variable? It is declared here, initialized on line 53, and then updated on line 73, but is otherwise never used or referenced anywhere else.

If you are trying to add views to the data in ADL, you'll need to define the view definition in the ADL config (`resources/integration_test/testdata/adl_db_config.json`), _not_ in the mongodb data.

We previously had this type of set up in an earlier iteration of the `DataLoader` (defining views in the `resources/integration_test/testdata/*.yml` files) but we realized that ADL doesn't distinguish between views and collections in mongodb. As in, if you have a mongodb datasource in ADL, ADL treats all views and collections from that mongodb as collections. The only views ADL acknowledges are those defined in the config.

Reply:
There is already a view in ADL, but when using it in the integration tests it is currently returning the error `MongoServerError: failed to query for tenant schema for "integration_test"."baz": schema not found`. We found out with Oliver that this seems to be an issue with the local ADL.
For integration tests, we are installing a new local ADL which has all the configuration for the views and collections we want to use. Next, the DataLoader will drop all the collections, add them back, populate them and generate/set their schema, before we start, using the Yaml file. Unfortunately, today, when a collection is dropped, retrieving the schema for the view fails, even after the collection schema is set.
So, I had to add a view section in the Yaml file matching the one in our ADL config to know there is a view we need to generate the schema for (but not load data into), before we can run the unit tests.

Reply:
As for the variable, I'm looking into it, I think you are right, it is not used.

Reply:
Ok that makes sense. So it looks like we don't even need a set that tracks all of the views, we just need to check if the dataset is a view during schema generation (a change already in this PR). Looks good 👍